dashboard/test/controllers/aichat_controller_test.rb

```diff
@@ -42,44 +50,44 @@ class AichatControllerTest < ActionController::TestCase
 
   test 'pilot teacher has access test' do
     sign_in(@genai_pilot_teacher)
-    post :chat_completion, params: @valid_params
+    post :chat_completion, params: @valid_params, as: :json
     assert_response :success
   end
 
   test 'pilot student has access test' do
     sign_in(@genai_pilot_student)
-    post :chat_completion, params: @valid_params
+    post :chat_completion, params: @valid_params, as: :json
     assert_response :success
   end
 
   test 'Bad request if required params are not included' do
     sign_in(@genai_pilot_teacher)
-    post :chat_completion, params: {newMessage: "hello"}
+    post :chat_completion, params: {newMessage: "hello"}, as: :json
     assert_response :bad_request
   end
 
   test 'Bad request if storedMessages param is not included' do
     sign_in(@genai_pilot_teacher)
-    post :chat_completion, params: @missing_stored_messages_params
+    post :chat_completion, params: @missing_stored_messages_params, as: :json
     assert_response :bad_request
   end
 
   # Post request with a profane messages param returns a failure
   test 'returns failure when chat message contains profanity' do
     sign_in(@genai_pilot_student)
     ShareFiltering.stubs(:find_failure).returns(ShareFailure.new(ShareFiltering::FailureType::PROFANITY, 'damn'))
-    post :chat_completion, params: @profanity_violation_params
-    assert_equal json_response["status"], ShareFiltering::FailureType::PROFANITY
-    assert_equal json_response["flagged_content"], "damn"
+    post :chat_completion, params: @profanity_violation_params, as: :json
+    assert_equal ShareFiltering::FailureType::PROFANITY, json_response["status"]
+    assert_equal "damn", json_response["flagged_content"]
```

Comment:
Argument order for `assert_equal` is expected, observed, so I switched to that. Makes failed tests easier to read :)